driver-core/src/main/com/mongodb/client/model/densify/DensifyRange.java

```diff
@@ -0,0 +1,164 @@
+import com.mongodb.client.model.MongoTimeUnit;
+import org.bson.BsonDocument;
+import org.bson.BsonInt64;
+import org.bson.BsonString;
+import org.bson.BsonType;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+import java.time.Instant;
+
+import static com.mongodb.assertions.Assertions.notNull;
+import static java.util.Arrays.asList;
+
+/**
+ * A specification of how to compute the missing {@linkplain Aggregates#densify(String, DensifyRange, DensifyOptions) field} values
+ * for which new documents must be added. It specifies a half-closed interval of values with the lower bound being inclusive, and a step.
+ * The first potentially missing value within each interval is its lower bound, other values are computed by adding the step
+ * multiple times, until the result is out of the interval. Each time the step is added, the result is a potentially missing value for
+ * which a new document must be added if the sequence of documents that is being densified does not have a document
+ * with equal value of the field.
+ *
+ * @see Aggregates#densify(String, DensifyRange, DensifyOptions)
+ * @see Aggregates#densify(String, DensifyRange)
+ * @mongodb.server.release 5.1
+ * @since 4.7
+ */
+@Evolving
+public interface DensifyRange extends Bson {
+    /**
+     * Returns a {@link DensifyRange} that represents an interval with the smallest
+     * BSON {@link BsonType#INT32 32-bit integer} / {@link BsonType#INT64 64-bit integer} / {@link BsonType#DOUBLE Double} /
+     * {@link BsonType#DECIMAL128 Decimal128} value of the {@linkplain Aggregates#densify(String, DensifyRange, DensifyOptions) field}
+     * in the sequence of documents being its lower bound, and the largest value being the upper bound.
+     *
+     * @param step The step.
+     * @return The requested {@link DensifyRange}.
+     */
+    static NumberDensifyRange fullRangeWithStep(final Number step) {
+        return new DensifyConstructibleBson(new Document("bounds", "full")
+                .append("step", notNull("step", step)));
+    }
+
+    /**
+     * Returns a {@link DensifyRange} that represents an interval with the smallest
+     * BSON {@link BsonType#INT32 32-bit integer} / {@link BsonType#INT64 64-bit integer} / {@link BsonType#DOUBLE Double} /
+     * {@link BsonType#DECIMAL128 Decimal128} value of the {@linkplain Aggregates#densify(String, DensifyRange, DensifyOptions) field}
+     * in the {@linkplain DensifyOptions#partitionByFields(Iterable) partition} of documents being its lower bound,
+     * and the largest value being the upper bound.
+     *
+     * @param step The step.
+     * @return The requested {@link DensifyRange}.
+     */
+    static NumberDensifyRange partitionRangeWithStep(final Number step) {
+        return new DensifyConstructibleBson(new Document("bounds", "partition")
+                .append("step", notNull("step", step)));
+    }
+
+    /**
+     * Returns a {@link DensifyRange} that represents a single interval [l; u).
```

Comment:
is `[l; u)` syntax for lower bound inclusive to upper bound?

Reply:
It's not, it's the standard interval notation: https://en.wikipedia.org/wiki/Interval_(mathematics). The documentation for `DensifyRange` also describes the interval in plain English: _"a half-closed interval of values with the lower bound being inclusive"_.

Reply:
Thanks, good to know - I just hadn't seen it with a semi-colon, just a comma. 

So its similar to the [maven ranges](https://docs.oracle.com/middleware/1212/core/MAVEN/maven_version.htm#MAVEN402).

Reply:
No worries - I learnt something new with the wiki page for intervals :)